controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -446,6 +448,65 @@ class ControllerManager : public rclcpp::Node
   };
 
   SwitchParams switch_params_;
+
+  class ControllerThreadWrapper // created this class so we can keep track of the time and period.
+  {
+  public:
+    explicit ControllerThreadWrapper( 
+       const rclcpp::Time & time, 
+       const rclcpp::Duration & period
+      )
+      : time_(time)
+      , period_(period)
+      , m_thread_{}
+    {
+    }
+    ControllerThreadWrapper(const ControllerThreadWrapper& t) = delete;
+    ControllerThreadWrapper(ControllerThreadWrapper&& t) = default;
+    ~ControllerThreadWrapper() {
+      if (m_thread_.joinable()) 
+      { 
+        m_thread_.join(); 
+      } 
+    }
+    
+    template< class... Args>
+    void start(Args&&... args)
+    {
+      m_thread_ = std::thread(std::forward<Args>(args)...);
+    }
+
+    void set_time_and_period(
+      const rclcpp::Time & time, 
+      const rclcpp::Duration & period) 
+    { 
+      time_ = time; 
+      period_ = period; 
+    }
+    
+    
+    rclcpp::Time get_time() const
+    { 
+      return time_; 
+    }
+    rclcpp::Duration get_period() const
+    { 
+      return period_; 
+    }
+
+    const char* k = "asdf";
+    // do we really need all these getters?
+    private:
+      rclcpp::Time time_;
```

Comment:
It doesn't really hurt at this point but I don't think these need to be members...we could keep the wrapper nevertheless, also useful if we want to report other info